server/svix-server/src/db/models/application.rs

```diff
@@ -42,21 +49,112 @@ impl Related<super::message::Entity> for Entity {
     }
 }
 
+impl Related<super::applicationmetadata::Entity> for Entity {
+    fn to() -> RelationDef {
+        Relation::Metadata.def()
+    }
+}
+
+impl Model {
+    pub async fn fetch_application_metadata(
+        &self,
+        db: &impl ConnectionTrait,
+    ) -> error::Result<applicationmetadata::ActiveModel> {
+        let query = applicationmetadata::Entity::secure_find(self.id.clone());
+        let metadata = ctx!(query.one(db).await)?
+            .map(applicationmetadata::ActiveModel::from)
+            .unwrap_or_else(|| applicationmetadata::ActiveModel::create(self.id.clone(), None));
+
+        Ok(metadata)
+    }
+
+    pub async fn fetch_many_with_metadata(
+        db: &DatabaseConnection,
+        org_id: OrganizationId,
+        limit: u64,
+        after_id: impl Into<Option<ApplicationId>>,
+    ) -> error::Result<impl Iterator<Item = (Self, Metadata)>> {
+        let mut query = Entity::secure_find(org_id)
+            .order_by_asc(Column::Id)
+            .limit(limit);
+
+        if let Some(id) = after_id.into() {
+            query = query.filter(Column::Id.gt(id))
+        }
+
+        let results = ctx!(
+            query
+                .find_also_related(applicationmetadata::Entity)
+                .all(db)
+                .await
+        )?;
+
+        Ok(results.into_iter().map(|(app, metadata)| {
+            let metadata = metadata.map(|m| m.metadata()).unwrap_or_default();
+            (app, metadata)
+        }))
+    }
+
+    pub async fn fetch_with_metadata(
```

Comment:
I can't get over this function and the next one. They are similarly named, no idea what one does vs the other, etc.

I'd just drop this one, it's so confusing.

Reply:
Updated, I think I was able to simplify things a bit